### PR TITLE
LIVE-2148: interactive atoms

### PR DESCRIPTION
**Pull request by joecowton1**
## Why are you doing this?

We're looking to include additional atoms and embeds in Editions where possible. This PR brings together the AR and ER approaches to CSP and allows us to render `interactive` atoms.

See this article for an example:
http://localhost:8080/world/2020/jun/10/uk-coronavirus-lockdown-20000-lives-boris-johnson-neil-ferguson?editions

## Changes

- Use `meta` component for Editions header
- share `csp` component between editions and AR
- add additional fields (rules?) to csp functions 

## Screenshots

| Before | After |
| --- | --- |
|  |  |

### CODE REVIEW

File: src/server/editionsPage.tsx [hunk 1]
```diff
@@ -34,37 +37,47 @@ const docParser = JSDOM.fragment.bind(null);
...
-		<meta name="viewport" content="initial-scale=1" />
-		<meta http-equiv="Content-Security-Policy" content="${cspString}" />
-        <style>${styles}</style>
+        ${renderToString(meta)}
+        <link rel="stylesheet" type="text/css" href="/fontSize.css">
+        <style>${generalStyles}</style>
         <style data-emotion-css="${emotionIds.join(' ')}">${itemStyles}</style>
+        <script id="targeting-params" type="application/json">
+            ${JSON.stringify(request.targetingParams)}
+        </script>
```
Comment (JamieB-gu) [inline src/server/editionsPage.tsx:78]: Is Editions starting to use the targeting params?

File: src/server/editionsPage.tsx [hunk 2]
```diff
@@ -102,7 +115,16 @@ function render(
...
-	const head = renderHead(request, thirdPartyEmbeds, body.css, body.ids);
+	const inlineStyles = requiresInlineStyles(request.content);
+
+	const head = renderHead(
+		item,
+		request,
+		thirdPartyEmbeds,
+		body.css,
+		body.ids,
+		inlineStyles,
```
Comment (JamieB-gu) [inline src/server/editionsPage.tsx:126]: I think you can just pass this as `false`, the Editions version of the CSP ignores it anyway.

File: src/server/editionsPage.tsx [hunk 3]
```diff
@@ -34,37 +37,47 @@ const docParser = JSDOM.fragment.bind(null);
...
+	const meta = h(Meta, { title: request.content.webTitle, cspString });
 
 	return `
-		<meta charSet="utf-8" />
-		<title>${request.content.webTitle}</title>
-		<meta name="viewport" content="initial-scale=1" />
-		<meta http-equiv="Content-Security-Policy" content="${cspString}" />
-        <style>${styles}</style>
+        ${renderToString(meta)}
+        <link rel="stylesheet" type="text/css" href="/fontSize.css">
```
Comment (JamieB-gu) [inline src/server/editionsPage.tsx:73]: Does Editions support the same scaling mechanism as Android? If not I think you can probably ignore this.

**Review by JamieB-gu — COMMENTED** [review]
Nice. Couple comments/questions.

File: src/server/editionsPage.tsx (hunk 1 above)
Comment (joecowton1) [reply src/server/editionsPage.tsx]: No don't think so, good call.

File: src/server/editionsPage.tsx (hunk 2 above)
Comment (joecowton1) [reply src/server/editionsPage.tsx]: Cool, fixed.

File: src/server/editionsPage.tsx (hunk 3 above)
Comment (joecowton1) [reply src/server/editionsPage.tsx]: Removed 👍

**Review by JamieB-gu — APPROVED** [review]

**Pull request merged**